lib/cocoapods/installer/analyzer.rb

```diff
@@ -406,6 +407,44 @@ def generate_target(target_definition, pod_targets)
         target
       end
 
+      # Verify that targets using a pod have the same swift version
+      #
+      # @param  [Hash{Podfile::TargetDefinition => Array<Specification>}] specs_by_target
+      #         the resolved specifications grouped by target.
+      #
+      # @note raises Informative if targets using a pod do not have
+      #       the same swift version
+      #
+      def check_pod_target_swift_versions(specs_by_target)
+        targets_by_spec = {}
+        swift_specs = Set.new
+        specs_by_target.each do |target, specs|
+          specs.each do |spec|
+            (targets_by_spec[spec] ||= []) << target
+          end
+          swift_specs.merge specs.select do |spec|
+            spec.consumer(target.platform).source_files.any? do |source_file|
```

Comment:
Yeah, I'm just about to push some commits that strip this down to just disallowing that, and strips swift version from PodVariant